src/Agent.Listener/Agent.cs

```diff
@@ -103,17 +103,37 @@ public override void Initialize(IHostContext hostContext)
 
                 if (parser.IsCommand("run") && !configManager.IsConfigured())
                 {
-                    throw new InvalidOperationException("Cannot run.  Must configure first.");
+                    throw new InvalidOperationException("Cannot run. Must configure first.");
                 }
 
                 Trace.Info("Done evaluating commands");
+                bool alreadyConfigured = configManager.IsConfigured();
                 await configManager.EnsureConfiguredAsync();
 
                 _inConfigStage = false;
 
-                // TODO: If configured as runasservice return from here
-            
-                return await RunAsync(TokenSource.Token);
+                AgentSettings settings = configManager.LoadSettings();
+                if (parser.IsCommand("run") || !settings.RunAsService)
+                {
+                    // Run the agent interactively
+                    Trace.Verbose(
+                        StringUtil.Format(
+                            "Run command mentioned: {0}, run as service option mentioned:{1}",
+                            parser.IsCommand("run"),
+                            settings.RunAsService));
+
+                    return await RunAsync(TokenSource.Token, settings);
+                }
+
+                if (alreadyConfigured)
+                {
+                    // This is helpful if the user tries to start the agent.listener which is already configured or running as service
+                    // However user can execute the agent by calling the run command
+                    // TODO: Should we check if the service is running and prompt user to start the service if its not already running?
+                    _term.WriteLine(StringUtil.Loc("ConfiguredAsRunAsService", settings.ServiceName));
```

Comment:
Its good to print the message that its configured as service along with the service name, otherwise if you start the agent.listener.exe it just terminates without any message in the terminal. I would prefer to keep the message as it will give some hint whats has happened.